src/Symfony/Component/HttpClient/Response/CurlResponse.php

```diff
@@ -258,8 +260,16 @@ private static function perform(CurlClientState $multi, array &$responses = null
             while (CURLM_CALL_MULTI_PERFORM === curl_multi_exec($multi->handle, $active));
 
             while ($info = curl_multi_info_read($multi->handle)) {
-                $multi->handlesActivity[(int) $info['handle']][] = null;
-                $multi->handlesActivity[(int) $info['handle']][] = \in_array($info['result'], [\CURLE_OK, \CURLE_TOO_MANY_REDIRECTS], true) || (\CURLE_WRITE_ERROR === $info['result'] && 'destruct' === @curl_getinfo($info['handle'], CURLINFO_PRIVATE)) ? null : new TransportException(sprintf('%s for "%s".', curl_strerror($info['result']), curl_getinfo($info['handle'], CURLINFO_EFFECTIVE_URL)));
+                $multi->handlesActivity[$id = (int) $info['handle']][] = null;
+                if (\in_array($info['result'], [\CURLE_OK, \CURLE_TOO_MANY_REDIRECTS], true) || (\CURLE_WRITE_ERROR === $info['result'] && 'destruct' === @curl_getinfo($info['handle'], CURLINFO_PRIVATE))) {
+                    $exception = null;
+                } elseif (CURLE_COULDNT_CONNECT === $info['result'] || (CURLE_OPERATION_TIMEOUTED === $info['result'] && 0 === curl_getinfo($info['handle'], CURLINFO_HTTP_CODE))) {
```

Comment:
this is too opinionated: CURLE_COULDNT_CONNECT can be triggered for many other reasons than a timeout I suppose